src/WebResponses.cpp

```diff
@@ -260,11 +255,29 @@ String AsyncWebServerResponse::_assembleHead(uint8_t version) {
 #endif
   out.concat(buf);
 
+  bool _sendAcceptsRange = true,
+    _sendTransferEncoding = true,
+    _sendContentType = true;
+
+  for (const auto& header : _headers) {
+    const char* headerName = header.name().c_str();
+    _sendAcceptsRange &= (strcasecmp(headerName, T_Accept_Ranges) == 0);
+    _sendTransferEncoding &= (strcasecmp(headerName, T_Transfer_Encoding) == 0);
+    if (!_sendAcceptsRange && !_sendTransferEncoding) break;
+  }
+
+  if (version) {
+    if (_sendAcceptsRange)
+      addHeader(T_Accept_Ranges, T_none);
+    if (_sendTransferEncoding && _chunked)
+      addHeader(T_Transfer_Encoding, T_chunked);
+  }
+
   if (_sendContentLength) {
     snprintf_P(buf, bufSize, PSTR("Content-Length: %d\r\n"), _contentLength);
     out.concat(buf);
   }
-  if (_contentType.length()) {
+  if (_sendContentType && _contentType.length()) {
```

Comment:
_sendContentType is never updated to false ? (No header check ?)

Reply:
My bad, forgot to include it. Now fixed!